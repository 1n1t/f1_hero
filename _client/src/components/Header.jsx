import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

class Header extends Component {
  renderUsername() {
    if (this.props.auth) {
      return (
        <li className="btn btn-secondary username">
          <i className="fa fa-user" aria-hidden="true" />
          {this.props.auth.username}
        </li>
      );
    }
  }

  render() {
    return (
      <header className="page-header">
        <h1 className="page-title">F1 Hero</h1>
        <nav className="navbar">
          <ul>
            <li>
              <Link to={'/picks'} className="btn btn-primary">
                My Picks
              </Link>
            </li>
            <li>
              <Link to={'/leaderboard'} className="btn btn-primary">
                Leaderboard
              </Link>
            </li>
            {this.renderUsername()}
          </ul>
        </nav>
      </header>
    );
  }
}

function mapStateToProps({ auth }) {
  return { auth };
}

export default connect(mapStateToProps)(Header);
