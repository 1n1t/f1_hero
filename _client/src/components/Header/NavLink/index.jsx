/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const NavLink = ({ path, name }) => (
	<Link to={path} className="btn btn-primary">
		{name}
	</Link>
);

NavLink.propTypes = {
	path: PropTypes.string.isRequired,
	name: PropTypes.string.isRequired
};

export default NavLink;
