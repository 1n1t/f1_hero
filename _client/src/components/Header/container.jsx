/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getCurrentUser } from '../../data/users/actions';
import Header from './index';

const mapStateToProps = state => ({ user: state.data.users.current, heading: state.services.page.heading });

// const HeaderContainer = connect(mapStateToProps, { getCurrentUser })(Header);

class HeaderContainer extends Component {
	componentDidMount() {
		this.props.getCurrentUser();
	}

	render() {
		return <Header user={this.props.user} heading={this.props.heading} />;
	}
}

export default connect(mapStateToProps, { getCurrentUser })(HeaderContainer);
