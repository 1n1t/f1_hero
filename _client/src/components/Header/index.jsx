import React from 'react';
import PropTypes from 'prop-types';

import NavLink from './NavLink';

const NavLinks = [
	{ path: '/picks', name: 'My Picks' },
	{ path: '/leaderboard', name: 'Leaderboard' },
	{ path: '/rules', name: 'Rules' }
];

const Header = ({ user }) => (
	<header className="page-header">
		<h1 className="page-title">
			F1 Hero <span>// Page header!!!</span>
		</h1>
		<nav className="navbar">
			<ul>
				{NavLinks.map(link => (
					<li>
						<NavLink {...link} key={link.name} />
					</li>
				))}
				<li className="btn btn-secondary username">
					<i className="fa fa-user" aria-hidden="true" />
					{user ? user.username : ''}
				</li>
			</ul>
		</nav>
	</header>
);

Header.propTypes = {
	user: PropTypes.shape({
		username: PropTypes.string
	}).isRequired
};

export default Header;
