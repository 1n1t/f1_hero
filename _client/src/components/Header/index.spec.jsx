import r from 'Root/jest/utils/renderer';
import userSample from 'Root/jest/samples/user';
import pageSample from 'Root/jest/samples/page';
import Header from './index';

const defaultProps = {
	user: userSample,
	heading: pageSample.heading
};

r.init(Header, defaultProps);

it('renders correctly', () => {
	const wrapper = r.render(null, true);
	expect(wrapper).toMatchSnapshot();
});
