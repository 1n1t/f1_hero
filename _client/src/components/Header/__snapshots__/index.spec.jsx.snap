// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<header
  className="Header"
>
  <h1
    className="page-title"
  >
    F1 Hero 
    <span>
      // 
      Foo Bar
    </span>
  </h1>
  <nav
    className="navbar"
  >
    <ul>
      <li
        key="My Picks"
      >
        <NavLink
          name="My Picks"
          path="/picks"
        />
      </li>
      <li
        key="Leaderboard"
      >
        <NavLink
          name="Leaderboard"
          path="/leaderboard"
        />
      </li>
      <li
        key="Rules"
      >
        <NavLink
          name="Rules"
          path="/rules"
        />
      </li>
      <li
        className="btn btn-secondary username"
      >
        <i
          aria-hidden="true"
          className="fa fa-user"
        />
        John Doe
      </li>
    </ul>
  </nav>
</header>
`;
